curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/review-algorithmic-thinking-by-building-a-dice-game/657e0c2c6a9d37705146f34d.md

```diff
@@ -45,19 +45,33 @@ assert.strictEqual(scoreSpans[5].innerText, ", score = 0");
 You should call your `detectFullHouse` function when your `rollDiceBtn` is clicked.
 
 ```js
-const counts = (arr) => arr.reduce((counts, num) => {counts[num] = ++counts[num] || 1; return counts}, {});
-const isFullHouse = (arr) => arr.includes(3) && arr.includes(2) && arr.length === 2;
-// Absurdly high, but will either timeout or bail early when assertion runs.
-for (let i = 0; i < 100_000; i++) {
-  rolls = 0;
-  rollDiceBtn.click();
-  if (isFullHouse(diceValuesArr)) {
-    assert.isFalse(scoreInputs[2].disabled);
-    assert.strictEqual(scoreInputs[2].value, "25");
-    assert.strictEqual(scoreSpans[2].innerText, ", score = 25");
-    break;
+//Temporarily modifies Math.random to guarentee full house.
+const math_random = Math.random
+const approx = 0.000001;
+const mymath_random = function*() {
+  yield 0;
+  yield 0;
+  yield 0;
+  yield 1/6 + approx;
+  yield 1/6 + approx;
+  while (true) {
+    yield math_random();
+  }
+}
+const mygen = mymath_random();
+
+try {
+  Math.random = () => {
+    return mygen.next().value;
   }
+  rollDiceBtn.click();
+} finally {
+  Math.random = math_random;
 }
```

Comment:
```suggestion
const origMathRandom = Math.random
const myMathRandom = (() => {
  let i = 0;
  const values = [0, 0, 0, 1 / 6 + 0.000001, 1 / 6 + 0.000001];
  return () => values[i++ % 5];
})();

Math.random = myMathRandom;
rollDiceBtn.click();
Math.random = origMathRandom;
```
proposed simpler version

Reply:
This version is definitely better, but I still think we need to use finally to reset Math.random to the original function. This is because rollDiceBtn.click() is edited by the user and might error.